dlib/CMakeLists.txt

```diff
@@ -502,6 +502,11 @@ if (NOT TARGET dlib)
          if (DLIB_USE_LAPACK)
             if (lapack_found)
                set (dlib_needed_libraries ${dlib_needed_libraries} ${lapack_libraries})
+               if (lapack_with_underscore)
+                  set (dlib_needed_definitions ${dlib_needed_definitions} -DLAPACK_FORCE_UNDERSCORE)
```

Comment:
You should add these things to the `active_preprocessor_switches` variable using `list(APPEND active_preprocessor_switches "-DWHATEVER")` rather than introducing a new variable.